torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -445,20 +452,28 @@ def _broadcast_params(self) -> None:
                 # Direct broadcasts only
                 for param in params:
                     if not self.should_bucket_param[i_param]:
-                        self.work_handles.append(
-                            dist.broadcast(tensor=param.data, src=global_src_rank, group=self.group, async_op=True)
+                        last_work_handle = dist.broadcast(
+                            tensor=param.data, src=global_src_rank, group=self.group, async_op=True
                         )
+
                     i_param += 1
 
                 # Bucket broadcasts
-                self.work_handles.append(
-                    dist.broadcast(tensor=bucket, src=global_src_rank, group=self.group, async_op=True)
-                )
+                if bucket.numel() > 0:
+                    last_work_handle = dist.broadcast(
+                        tensor=bucket, src=global_src_rank, group=self.group, async_op=True
+                    )
 
-        # Consume all async calls
-        while len(self.work_handles) > 0:
```

Comment:
we can just wait on the last call actually, that was not wrong but not useful and slower